player/playloop.c

```diff
@@ -591,7 +595,9 @@ int get_current_chapter(struct MPContext *mpctx)
     for (i = 0; i < mpctx->num_chapters; i++)
         if (current_pts < mpctx->chapters[i].pts)
             break;
-    return MPMAX(mpctx->last_chapter_seek, i - 1);
+    i--;
+    return mpctx->last_chapter_flag ?
+        mpctx->last_chapter_seek : MPMAX(mpctx->last_chapter_seek, i);
```

Comment:
Accidental leftover from debugging? Surely this can just be `i-1` in the return.

Reply:
I did that because `MPMAX(a, b)` expands into a ternary statement. Either `a` or `b` is going to be computed twice, once for the evaluation, and again for the result.

This can be avoided by using an inline function, or by using a macro implementation like the one GCC recommends:
```c
#define maxint(a,b) \
  ({int _a = (a), _b = (b); _a > _b ? _a : _b; })
```
but then you'll need separate macros/functions for ints and floats. `MPMAX` is all-purpose, which is nice, but we should avoid performing operations inside of its parameters.

Reply:
The attention to avoiding an extra operation is appreciated, but for readability's sake,  I'd rather just have `MPMAX(mpctx->last_chapter_seek, i - 1);` since the whole point of the macro is convenience. Plenty of places already do operations within MPMAX or MPMIN and it's not like this is performance critical code. Maybe `MPMAX` and `MPMIN` could be improved later.